src/psyclone/tests/domain/common/extract_driver_creator_test.py

```diff
@@ -250,16 +249,19 @@ def test_rename_suffix_if_name_clash():
     # contain out_fld_post for the input variable out_fld_post,
     # and "out_fld_post0" for the output value of out_fld.
     expected = """
-      CALL extract_psy_data%PreDeclareVariable("out_fld_post", out_fld_post)
-      CALL extract_psy_data%PreDeclareVariable("in_out_fld_post0", in_out_fld)
-      CALL extract_psy_data%PreDeclareVariable("out_fld_post0", out_fld)
-      CALL extract_psy_data%ProvideVariable("in_out_fld", in_out_fld)
-      CALL extract_psy_data%ProvideVariable("out_fld_post", out_fld_post)
-      CALL extract_psy_data%ProvideVariable("in_out_fld_post0", in_out_fld)
-      CALL extract_psy_data%ProvideVariable("out_fld_post0", out_fld)"""
+      CALL extract_psy_data % PreDeclareVariable("out_fld_post", out_fld_post)
+      CALL extract_psy_data % PreDeclareVariable("in_out_fld_post0", """\
+      """in_out_fld)
+      CALL extract_psy_data % PreDeclareVariable("out_fld_post0", out_fld)
+      CALL extract_psy_data % PreDeclareVariable("out_fld", out_fld)
+      CALL extract_psy_data % ProvideVariable("in_out_fld", in_out_fld)
+      CALL extract_psy_data % ProvideVariable("out_fld", out_fld)
+      CALL extract_psy_data % ProvideVariable("out_fld_post", out_fld_post)
+      CALL extract_psy_data % ProvideVariable("in_out_fld_post0", in_out_fld)
+      CALL extract_psy_data % ProvideVariable("out_fld_post0", out_fld)"""
```

Comment:
I'm a bit confused here. How come we now write the values of `out_fld` for both "out_fld" and "out_fld_post0"?

Reply:
How should I know :) :) :)
OK, I've added a serious amount of comments to explain that. I hope it's better now.

Reply:
Awesome, thanks Joerg.